From 32d9819cdb89e17f0b58556ea32d5a08113db4a8 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Thu, 22 Aug 2019 15:44:15 +0200
Subject: [PATCH] Make parts of VideoCaptureSettings and AudioCaptureSettings
 move constructible/assignable

VideoCaptureSettings and AudioCaptureSettings move constructors and assign operators have
been declared noexcept using default implementation. This requires all their parts to be
move constructible/assignable.

Bug: 819294
Change-Id: Ic75aeb92419661a342f46bde07789f6bae3f5162

diff --git a/third_party/blink/public/platform/modules/mediastream/media_stream_audio_processor_options.h b/third_party/blink/public/platform/modules/mediastream/media_stream_audio_processor_options.h
index e3740ec7ac0d..4d59b18f7f07 100644
--- a/third_party/blink/public/platform/modules/mediastream/media_stream_audio_processor_options.h
+++ b/third_party/blink/public/platform/modules/mediastream/media_stream_audio_processor_options.h
@@ -55,7 +55,10 @@ struct BLINK_PLATFORM_EXPORT AudioProcessingProperties {
   // their default values.
   AudioProcessingProperties();
   AudioProcessingProperties(const AudioProcessingProperties& other);
+  AudioProcessingProperties(AudioProcessingProperties&& other) noexcept;
   AudioProcessingProperties& operator=(const AudioProcessingProperties& other);
+  AudioProcessingProperties& operator=(
+      AudioProcessingProperties&& other) noexcept;
 
   // Disables properties that are enabled by default.
   void DisableDefaultProperties();
diff --git a/third_party/blink/public/web/modules/mediastream/video_track_adapter_settings.h b/third_party/blink/public/web/modules/mediastream/video_track_adapter_settings.h
index e34a78d5acfa..366e2e93d512 100644
--- a/third_party/blink/public/web/modules/mediastream/video_track_adapter_settings.h
+++ b/third_party/blink/public/web/modules/mediastream/video_track_adapter_settings.h
@@ -31,7 +31,10 @@ class BLINK_MODULES_EXPORT VideoTrackAdapterSettings {
                             double max_aspect_ratio,
                             double max_frame_rate);
   VideoTrackAdapterSettings(const VideoTrackAdapterSettings& other);
+  VideoTrackAdapterSettings(VideoTrackAdapterSettings&& other) noexcept;
   VideoTrackAdapterSettings& operator=(const VideoTrackAdapterSettings& other);
+  VideoTrackAdapterSettings& operator=(
+      VideoTrackAdapterSettings&& other) noexcept;
   bool operator==(const VideoTrackAdapterSettings& other) const;
 
   const base::Optional<gfx::Size>& target_size() const { return target_size_; }
diff --git a/third_party/blink/renderer/modules/mediastream/video_track_adapter_settings.cc b/third_party/blink/renderer/modules/mediastream/video_track_adapter_settings.cc
index b23e06d6cc68..e8ff3f90522e 100644
--- a/third_party/blink/renderer/modules/mediastream/video_track_adapter_settings.cc
+++ b/third_party/blink/renderer/modules/mediastream/video_track_adapter_settings.cc
@@ -42,8 +42,12 @@ VideoTrackAdapterSettings::VideoTrackAdapterSettings(
 
 VideoTrackAdapterSettings::VideoTrackAdapterSettings(
     const VideoTrackAdapterSettings& other) = default;
+VideoTrackAdapterSettings::VideoTrackAdapterSettings(
+    VideoTrackAdapterSettings&& other) noexcept = default;
 VideoTrackAdapterSettings& VideoTrackAdapterSettings::operator=(
     const VideoTrackAdapterSettings& other) = default;
+VideoTrackAdapterSettings& VideoTrackAdapterSettings::operator=(
+    VideoTrackAdapterSettings&& other) noexcept = default;
 
 bool VideoTrackAdapterSettings::operator==(
     const VideoTrackAdapterSettings& other) const {
diff --git a/third_party/blink/renderer/platform/exported/mediastream/media_stream_audio_processor_options.cc b/third_party/blink/renderer/platform/exported/mediastream/media_stream_audio_processor_options.cc
index 86fdef71d90f..91d34290f985 100644
--- a/third_party/blink/renderer/platform/exported/mediastream/media_stream_audio_processor_options.cc
+++ b/third_party/blink/renderer/platform/exported/mediastream/media_stream_audio_processor_options.cc
@@ -78,8 +78,12 @@ void GetExtraConfigFromJson(
 AudioProcessingProperties::AudioProcessingProperties() = default;
 AudioProcessingProperties::AudioProcessingProperties(
     const AudioProcessingProperties& other) = default;
+AudioProcessingProperties::AudioProcessingProperties(
+    AudioProcessingProperties&& other) noexcept = default;
 AudioProcessingProperties& AudioProcessingProperties::operator=(
     const AudioProcessingProperties& other) = default;
+AudioProcessingProperties& AudioProcessingProperties::operator=(
+    AudioProcessingProperties&& other) noexcept = default;
 
 void AudioProcessingProperties::DisableDefaultProperties() {
   echo_cancellation_type = EchoCancellationType::kEchoCancellationDisabled;
-- 
2.20.1

